include/openthread/platform/ble.h

```diff
@@ -0,0 +1,266 @@
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file defines a OpenThread BLE GATT peripheral interface driver.
+ *
+ */
+
+#ifndef OT_PLATFORM_BLE_H_
+#define OT_PLATFORM_BLE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+#include <openthread/error.h>
+#include <openthread/instance.h>
+
+/**
+ * @addtogroup plat-ble
+ *
+ * @brief
+ *   This module includes the platform abstraction for BLE Host communication.
+ *   The platform needs to implement Bluetooth LE 4.2 or higher.
+ *
+ * @{
+ *
+ */
+
+enum
+{
+    /**
+     * Time slot duration on PHY layer in microseconds (0.625ms).
+     */
+    OT_BLE_TIMESLOT_UNIT = 625,
+
+    /**
+     * Minimum allowed interval for advertising packet in OT_BLE_ADV_INTERVAL_UNIT units (20ms).
+     */
+    OT_BLE_ADV_INTERVAL_MIN = 0x0020,
+
+    /**
+     * Maximum allowed interval for advertising packet in OT_BLE_ADV_INTERVAL_UNIT units (10.24s).
+     */
+    OT_BLE_ADV_INTERVAL_MAX = 0x4000,
+
+    /**
+     * Unit used to calculate interval duration (0.625ms).
+     */
+    OT_BLE_ADV_INTERVAL_UNIT = OT_BLE_TIMESLOT_UNIT,
+};
+
+/**
+ * This structure represents an BLE packet.
+ *
+ */
+typedef struct otBleRadioPacket
+{
+    uint8_t *mValue;  ///< The value of an attribute
+    uint16_t mLength; ///< Length of the @p mValue.
+    int8_t   mPower;  ///< Transmit/receive power in dBm.
+} otBleRadioPacket;
+
+/*******************************************************************************
+ * @section Bluetooth Low Energy management.
+ ******************************************************************************/
+
+/**
+ * Enable the Bluetooth Low Energy radio.
+ *
+ * @note BLE Device should use the highest ATT_MTU supported that does not
+ * exceed OT_BLE_ATT_MTU_MAX octets.
```

Comment:
OT_BLE_ATT_MTU_MAX should be defined in this header and kPacketBufferSize in BleSecure.hpp should be set accordingly. Will take care. 
Alternatively, it may become a configurable performance parameter.

Reply:
fixed